docs/gitlab/gitlab-aws.md

```diff
@@ -0,0 +1,537 @@
+1. Choose Roles, then Create role.
+1. Choose EKS from the list of services, then "Allows EKS to manage clusters on your behalf", then Next: Permissions.
+1. Choose Next: Tags.
+1. Choose Next: Review.
+1. For the Role name, enter a unique name for your role, such as eksServiceRole, then choose Create role.
+1. Set the EKS_ROLE_ARN environment variable to the name of your role ARN
+    ```console
+    export EKS_ROLE_ARN=replace-with-full-role-arn
+    ```
+
+#### Create your Amazon EKS Cluster VPC
+[Original Source Guide](https://docs.aws.amazon.com/eks/latest/userguide/getting-started.html)
+
+1. Open the [AWS CloudFormation console](https://console.aws.amazon.com/cloudformation).
+1. From the navigation bar, select a Region that supports Amazon EKS.
+    > Note: Amazon EKS is available in the following Regions at this time:
+     > * US West (Oregon) (us-west-2)
+     > * US East (N. Virginia) (us-east-1)
+     > * EU (Ireland) (eu-west-1)
+
+1. Set the REGION environment variable to your region
+    ```console
+    export REGION=replace-with-region
+    ```
+
+1. Choose Create stack.
+1. For Choose a template, select Specify an Amazon S3 template URL.
+1. Paste the following URL into the text area and choose Next.
+    ```
+    https://amazon-eks.s3-us-west-2.amazonaws.com/cloudformation/2018-11-07/amazon-eks-vpc-sample.yaml
+    ```
+1. On the Specify Details page, fill out the parameters accordingly, and choose Next.
+    ```
+    * Stack name: Choose a stack name for your AWS CloudFormation stack. For example, you can call it eks-vpc.
+    * VpcBlock: Choose a CIDR range for your VPC. You may leave the default value.
+    * Subnet01Block: Choose a CIDR range for subnet 1. You may leave the default value.
+    * Subnet02Block: Choose a CIDR range for subnet 2. You may leave the default value.
+    * Subnet03Block: Choose a CIDR range for subnet 3. You may leave the default value.
+    ```
+1. (Optional) On the Options page, tag your stack resources and choose Next.
+1. On the Review page, choose Create.
+1. When your stack is created, select it in the console and choose Outputs.
+1. Using values from outputs, export the following environment variables.
+    ```console
+    export EKS_VPC=replace-with-eks-vpcId
+    export EKS_SUBNETS=replace-with-eks-subnetIds01,replace-with-eks-subnetIds02,replace-with-eks-subnetIds03
+    export EKS_SECURITY_GROUP=replace-with-eks-securityGroupId
+    ```
+
+#### Create an RDS subnet group
+1. Navigate to the aws console in same region as the EKS cluster
+1. Navigate to `RDS` service
+1. Navigate to `Subnet groups` in left hand pane
+1. Click `Create DB Subnet Group`
+1. Name your subnet i.e. `eks-db-subnets`
+1. Select the VPC created in the EKS VPC step
+1. Click `Add all subnets related to this VPC`
+1. Click Create
+1. Export the db subnet group name
+    ```console
+    export RDS_SUBNET_GROUP_NAME=replace-with-DBSubnetgroup-name
+    ```
+#### Create an RDS Security Group (example only)
```

Comment:
Can we use the default security group of the VPC we created to put the EKS cluster in?